app/notifications/process_notifications.py

```diff
@@ -156,36 +156,37 @@ def send_notification_to_queue(
 ):
     """
     Create, enqueue, and asynchronously execute a Celery task to send a notification.
+    This is the execution path for sending notifications with contact information.
     """
 
-    deliver_task, queue = _get_delivery_task(notification, research_mode, queue, sms_sender_id)
+    tasks = []
 
-    # This is a relationship to a TemplateHistory instance.
-    template = notification.template
+    if recipient_id_type:
+        # This is a relationship to a TemplateHistory instance.
+        template = notification.template
 
-    if template:
         # This is a nullable foreign key reference to a CommunicationItem instance UUID.
-        communication_item_id = template.communication_item_id
+        communication_item_id = template.communication_item_id if template else None
 
-    try:
-        # Including sms_sender_id is necessary so the correct sender can be chosen.
-        # https://docs.celeryq.dev/en/v4.4.7/userguide/canvas.html#immutability
-        tasks = [deliver_task.si(str(notification.id), sms_sender_id).set(queue=queue)]
-        if recipient_id_type and communication_item_id:
-            tasks.insert(
-                0,
+        if communication_item_id is not None:
+            if recipient_id_type != IdentifierType.VA_PROFILE_ID.value:
+                tasks.append(lookup_va_profile_id.si(notification.id).set(queue=QueueNames.LOOKUP_VA_PROFILE_ID))
+
+            tasks.append(
                 lookup_recipient_communication_permissions.si(str(notification.id)).set(
                     queue=QueueNames.COMMUNICATION_ITEM_PERMISSIONS
-                ),
+                )
             )
 
-            if recipient_id_type != IdentifierType.VA_PROFILE_ID.value:
-                tasks.insert(0, lookup_va_profile_id.si(notification.id).set(queue=QueueNames.LOOKUP_VA_PROFILE_ID))
+    # Including sms_sender_id is necessary so the correct sender can be chosen.
+    # https://docs.celeryq.dev/en/v4.4.7/userguide/canvas.html#immutability
+    deliver_task, queue = _get_delivery_task(notification, research_mode, queue, sms_sender_id)
+    tasks.append(deliver_task.si(str(notification.id), sms_sender_id).set(queue=queue))
 
+    try:
         # This executes the task list.  Each task calls a function that makes a request to
         # the backend provider.
         chain(*tasks).apply_async()
-
     except Exception:
         dao_delete_notification_by_id(notification.id)
         raise
```

Comment:
whoa. We are deleting notifications here? I guess I never realized that. So if an execption is thrown in the chain we just delete it?

Reply:
I wondered about that too.  I think this catches an exception when trying to pass the Celery task chain to Celery.  An exception here indicates that no tasks where enqueued.